### ISSUE TITLE
Improve 'rate this app'

### ISSUE THREAD

**Issue by Nuytemans-Dieter** [TNelen/BlackBox#196]
First show a popup that allows the user to rate the app. Preferably gather this data using Firebase analytics as well.
Only ratings of 4/5 stars will be sent to the Play store, the others will find contact information to get support.

**Comment by Nuytemans-Dieter** [comment]
Forgot to mention the issue in 80c5b7f7f8af010a3ca0846856cac9e6a74fca9b.
Not using the API yet, but rating behavior has been improved.

**Comment by TNelen** [comment]
Putting this issueon hold  for now. This is good enough; Api integration is fun project in the future but not needed for now:

**Comment by TNelen** [comment]
Added a contact us button which redirects to a mail app to ask for support.

**Comment by TNelen** [comment]
Reopened issue:
We need to add this functionality to the popup:
https://medium.com/@naveentp/integrating-in-app-review-api-into-android-app-aeea0edc62f2

In this way user does not have to leave the app, but an rate the app ingame (do this only for 3 or 4+ start obviously :p )

**Comment by Nuytemans-Dieter** [comment]
The design guidelines and restrictions do mention that this should not be linked to a button click (or in fact, even after asking their opinion of the app) but rather when a user has experienced enough of the app. Should we add this anyway on button click?

Alternatives:
- Keep track of number of questions played, show this after question n
- (Also on button click) Only show the rate button on home screen after playing n questions but without our own, custom stars popup

Any other ideas?

**Comment by TNelen** [comment]
Duplicate of #217

**Comment by TNelen** [comment]
use this library 
https://pub.dev/packages/rate_my_app

**Comment by TNelen** [comment]
Rate this app popup is show in partymode when there are 10 questions remaining.

**Comment by TNelen** [comment]
Maybe make this smarter in the future. I.e. after spending x time in app, of opening app Y times.

**Comment by Nuytemans-Dieter** [comment]
> Maybe make this smarter in the future. I.e. after spending x time in app, of opening app Y times.

Good idea! It would be interesting to ask their opinion when we believe they are enjoying the app.

**Comment by TNelen** [comment]
We NEEED to check out this! 
https://pub.dev/packages/in_app_review
